Source/Core/Core/PowerPC/JitArm64/Jit.cpp

```diff
@@ -429,20 +464,24 @@ void JitArm64::WriteExit(Arm64Gen::ARM64Reg dest, bool LK, u32 exit_address_afte
   }
 }
 
-void JitArm64::FakeLKExit(u32 exit_address_after_return)
+void JitArm64::FakeLKExit(u32 exit_address_after_return, ARM64Reg preloaded_after_reg)
 {
   if (!m_enable_blr_optimization)
     return;
 
   // We may need to fake the BLR stack on inlined CALL instructions.
   // Else we can't return to this location any more.
   gpr.Lock(ARM64Reg::W30);
-  ARM64Reg after_reg = gpr.GetReg();
+  ARM64Reg after_reg = preloaded_after_reg;
+  if (after_reg == ARM64Reg::INVALID_REG)
+    after_reg = gpr.GetReg();
   ARM64Reg code_reg = gpr.GetReg();
-  MOVI2R(after_reg, exit_address_after_return);
   ADR(EncodeRegTo64(code_reg), 12);
   STP(IndexType::Pre, EncodeRegTo64(code_reg), EncodeRegTo64(after_reg), ARM64Reg::SP, -16);
-  gpr.Unlock(after_reg, code_reg);
+  if (after_reg != preloaded_after_reg)
+    gpr.Unlock(after_reg, code_reg);
+  else
+    gpr.Unlock(code_reg);
```

Comment:
Simpler to express as:

```suggestion
  gpr.Unlock(code_reg);
  if (after_reg != preloaded_after_reg)
    gpr.Unlock(after_reg);
```